scripts/deploy.js

```diff
@@ -47,13 +46,16 @@ const deployToEdge = async () => {
     clientSecret: EDGE_CLIENT_SECRET,
     accessTokenUrl: EDGE_ACCESS_TOKEN_URL,
   });
-
-  const publishResp = await edgeStore.submit({
-    filePath: ZIP_PATH,
-    notes: 'Updating extension.',
-  });
-
-  console.log('publishResp: ', publishResp);
+  try {
+    const publishResp = await edgeStore.submit({
+      filePath: ZIP_PATH,
+      notes: 'Updating extension.',
+    });
+    console.log('Edge publishResp:', publishResp);
+  } catch (error) {
+    console.error('Edge Deployment failed:', error);
+    throw error;
```

Comment:
Do we need to throw again because this throw will also dump lots of information in the log.

Reply:
And also, directly log `error` object may not show all the details.

Reply:
The return of `publishResp` is a string. I used `try... catch...`  here to obtain the reason for the error, while also wanting to terminate the action operation in case of an error, causing the action operation to fail. I haven't really thought of a better way here. Do you have any better suggestions?

Reply:
Actually there is no code after `catch`, so just `return` or even just `log` is enough. But without throw, the actions won't report error and seems like it is successful right?

To terminate the workflow and let actions to know that the workflow has failed, we may just use `process.exit(-1)` to exit and `-1` means there are errors in it.

The error should be an exception object in js, so maybe can use `error.message` to obtain the reason.